generic3g/actions/AccumulatorActionInterface.F90

```diff
@@ -35,33 +38,35 @@ module mapl3g_AccumulatorActionInterface
    logical function accumulation_type_is_valid(acctype) result(lval)
       character(len=*), optional, intent(in) :: acctype
 
-      lval = present(acctype)
-      if(lval) lval = any(ACCUMULATION_TYPES == acctype)
+      lval = .FALSE.
+      if(.not. present(acctype)) return
+      lval = any(ACCUMULATION_TYPES == acctype)
 
    end function accumulation_type_is_valid
 
    subroutine get_accumulator_action(accumulation_type, typekind, action, rc)
       character(len=*), intent(in) :: accumulation_type
       type(ESMF_TypeKind_Flag), intent(in) :: typekind
-      class(ExtensionAction), allocatable, intent(out) :: action
+      class(ExtensionAction), allocatable, intent(inout) :: action
```

Comment:
Really this is intent OUT.  Indeed, were it not for return codes (and compiler bugs), I'd want this to be a function instead of a subroutine.

```suggestion
      class(ExtensionAction), allocatable, intent(out) :: action
```

Reply:
Done